godot_project/autoloads/openmm/openmm_payload.gd

```diff
@@ -372,6 +432,22 @@ func _passivate_atom(in_structure: AtomicStructure, in_atom_id: int) -> PackedVe
 	return directions
 
 
+func _calculate_remaining_valence(in_structure: AtomicStructure, in_atom_id: int, in_included_bond_ids: PackedInt32Array) -> int:
+	var data: ElementData = PeriodicTable.get_by_atomic_number(in_structure.atom_get_atomic_number(in_atom_id))
+	var atom_bonds: PackedInt32Array = in_structure.atom_get_bonds(in_atom_id)
+	var used_valence: int = 0
+	for bond_id in atom_bonds:
+		if not bond_id in in_included_bond_ids:
+			continue
+		var bond_order: int = in_structure.get_bond(bond_id).z
+		used_valence += bond_order
+	var valence_left: int = data.valence
+	if data.number > 5:
+		valence_left = 8 - valence_left
+	valence_left -= used_valence
+	return valence_left
+
+
```

Comment:
Notes for a future PR: 
There is something similar in `add_posing_atom_input_handler.gd`, (mainly, the diff between `get_charge()` and `get_stable_charge()`) but also in `ring_action_add_hydrogens.gd`

It would be great to have a single place where we do these calculations instead of having 3 different implementations for the same thing.